packages/stencil/components/select/src/components/osds-select/osds-select.tsx

```diff
@@ -253,6 +253,13 @@ export class OsdsSelect implements OdsSelect<OdsStencilMethods<OdsSelectMethods>
     }
     this.logger.log('[checkForClickOutside]', arguments, { validity: this.validityState });
     this.controller.closeSurface();
+
+    const selectOptions = this.el.querySelectorAll('osds-select-option');
```

Comment:
can be done in a single loop to increase perf, something like :
```
this.el.querySelectorAll('osds-select-option').forEach((option) => {
  if (this.value && option.getAttribute('value') === this.value) {
    option.setAttribute('selected', 'true');
  } else {
    option.setAttribute('selected', 'false');
  }
})
```

Reply:
You can also get the list of osds-select-options un the controller with `this.controller.selectOptions`

Reply:
Fixed !